apps/staking/src/components/OracleIntegrityStaking/index.tsx

```diff
@@ -98,46 +119,270 @@ export const OracleIntegrityStaking = ({
           { "mt-6": self === undefined },
         )}
       >
-        <div className="relative w-full overflow-x-auto">
-          <h3 className="sticky left-0 mb-4 pl-4 text-2xl font-light sm:pb-4 sm:pl-10 sm:pt-6">
-            {self ? "Other Publishers" : "Publishers"}
-          </h3>
+        <PublisherList
+          title={self ? "Other Publishers" : "Publishers"}
+          availableToStake={availableToStake}
+          publishers={otherPublishers}
+          totalStaked={staked}
+        />
+      </div>
+    </ProgramSection>
+  );
+};
+
+type PublisherListProps = {
+  title: string;
+  availableToStake: bigint;
+  totalStaked: bigint;
+  publishers: PublisherProps["publisher"][];
+};
+
+const PublisherList = ({
+  title,
+  availableToStake,
+  publishers,
+  totalStaked,
+}: PublisherListProps) => {
+  const [search, setSearch] = useState("");
+  const [sort, setSort] = useState({
+    field: SortField.PoolUtilization,
+    descending: false,
+  });
+  const filter = useFilter({ sensitivity: "base", usage: "search" });
+  const [currentPage, setPage] = useState(0);
+  const filteredSortedPublishers = useMemo(() => {
+    const sorted = publishers
+      .filter(
+        (publisher) =>
+          filter.contains(publisher.publicKey, search) ||
+          (publisher.name !== undefined &&
+            filter.contains(publisher.name, search)),
+      )
+      .sort((a, b) => {
+        switch (sort.field) {
+          case SortField.PublisherName: {
+            return (a.name ?? a.publicKey).localeCompare(b.name ?? b.publicKey);
+          }
+          case SortField.LastEpochAPY: {
+            return (
+              calculateApy(a.poolCapacity, a.poolUtilization, false) -
+              calculateApy(b.poolCapacity, b.poolUtilization, false)
+            );
+          }
+          case SortField.NumberOfFeeds: {
+            return Number(a.numFeeds - b.numFeeds);
+          }
+          case SortField.PoolUtilization: {
+            return Number(
+              a.poolUtilization / a.poolCapacity -
+                b.poolUtilization / b.poolCapacity,
```

Comment:
you can re-write a/b < c/d as a*d < c*d to avoid divisions thus being less error prone.